src/intrinsics.cpp

```diff
@@ -432,6 +450,19 @@ static jl_cgval_t emit_runtime_call(jl_codectx_t &ctx, JL_I::intrinsic f, const
     return mark_julia_type(ctx, r, true, (jl_value_t*)jl_any_type);
 }
 
+static jl_cgval_t emit_runtime_intrinsic(jl_codectx_t &ctx, jl_cgval_t *f, const jl_cgval_t *argv, size_t nargs)
+{
+    JL_I::intrinsic i = (JL_I::intrinsic)*(uint32_t*)f;
+    JuliaFunction *fptr = (JuliaFunction*)jl_intrinsic_table.fptrs[i];
```

Comment:
This is a `jl_value_t* (*)(jl_value_t*, ...)`, you cannot simply cast a function to a C++ class.

Reply:
Oops, yeah, I need a different way of packaging this into the call

Reply:
yes, see `emit_runtime_call`, which has those JuliaFunction objects